test/addons-napi/test_promise/test_promise.c

```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+#include <node_api.h>
+#include "../common.h"
+
+napi_value bind_napi_create_promise(napi_env env, napi_callback_info info) {
+  napi_value promise, resolve, reject, result;
+
+  NAPI_CALL(env, napi_create_promise(env, &promise, &resolve, &reject));
+  NAPI_CALL(env, napi_create_object(env, &result));
+
+  fprintf(stderr, "resolve: %p\n", resolve);
+
+  napi_valuetype value_type;
+  NAPI_CALL(env, napi_typeof(env, resolve, &value_type));
+  fprintf(stderr, "resolve: value_type: %d\n", value_type);
+
+  napi_property_descriptor properties[] = {
+    { "promise", NULL, NULL, NULL, NULL, promise, napi_default, NULL },
+    { "resolve", NULL, NULL, NULL, NULL, resolve, napi_default, NULL },
+    { "reject", NULL, NULL, NULL, NULL, reject, napi_default, NULL },
+  };
```

Comment:
The test uses the pattern where the promise is created in the native code and the resolve, reject are called through JS.  If this is the most common use case I wonder a helper  (or simply returning it from the original call) to create JS object would make any sense.